src/include/migraphx/op/scatter.hpp

```diff
@@ -38,30 +46,41 @@ struct scatter
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
         check_shapes{inputs, *this}.has(3).standard();
-        return inputs.front();
+        // If non-packed, this converts to a packed output while preserving permutation of tensor
+        return inputs.front().with_lens(inputs.front().lens());
     }
 
     argument compute(const shape& output_shape, std::vector<argument> args) const
     {
         argument result{output_shape};
+        auto& self = static_cast<const Derived&>(*this);
+
         // max dimension in axis
         auto axis_dim_size = output_shape.lens()[axis];
+        // iterate through all elements in output
         visit_all(result, args[0], args[2])([&](auto output, auto data, auto update) {
             std::copy(data.begin(), data.end(), output.begin());
             args[1].visit([&](auto indices) {
                 auto ind_s = indices.get_shape();
+                // iterate through items in index
                 shape_for_each(ind_s, [&](const auto& idx) {
-                    auto out_idx  = idx;
-                    auto index    = indices[ind_s.index(idx)];
+                    auto out_idx = idx;
+                    auto index   = indices[ind_s.index(idx)];
+                    // normalize negative indexes
                     index         = (index < 0) ? index + axis_dim_size : index;
                     out_idx[axis] = index;
-                    output[output_shape.index(out_idx)] = update[ind_s.index(idx)];
+
+                    // call reduction() method of derived struct
+                    self.reduction()(output[output_shape.index(out_idx)], update[ind_s.index(idx)]);
                 });
             });
         });
 
         return result;
     }
+    auto init() const {}
+    scatter() {}
+    scatter(int64_t ax) : axis(ax) {}
```

Comment:
I cant comment on the line in github, but the `name()` method in this class should be removed.